designs/NuGetExtensibility/INuGetProjectServices.md

```diff
@@ -0,0 +1,153 @@
+# INuGetProjectServices
+
+* Status: In Review
+* Author(s): [Andy Zivkovic](https://github.com/zivkan)
+* Issue: [9577](https://github.com/NuGet/Home/issues/9577) IVsPackageInstallerServices should be async
+
+## Problem Background
+
+The existing IVsPackageInstallerServices service is synchronous, and is being called by several components on the UI thread, causing UI delays in Visual Studio. We are creating new async APIs that will allow us to unblock the UI thread, and use the Visual Studio Service Broker to work in Visual Studio CodeSpaces (and maybe Live Share).
+
+See [NuGet extensibility services](NuGetExtensibilityServices.md) for more details.
+
+## Who are the customers
+
+Visual Studio component and extension developers.
+
+## Goals
+
+* Can be called on any thread (ideally background thread, but work if called on UI thread).
+* Do not block caller (implement async properly, return incomplete task instead).
+* Works with Visual Studio Codespaces, and hopefully Live Share.
```

Comment:
Can I make a console application that leverages this API? Should this be a goal to support this scenario?

Reply:
No, this is the contract for extensibility that we provide within Visual Studio.

Reply:
@zivkan answered more of my questions offline, closing this off now